src/test/java/org/llorllale/cactoos/matchers/HasValuesTest.java

```diff
@@ -42,81 +40,74 @@
  * @since 1.0.0
  * @checkstyle MagicNumberCheck (500 lines)
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle RegexpSinglelineCheck (500 lines)
  * @checkstyle MultilineJavadocTagsCheck (500 lines)
+ * @checkstyle StringLiteralsConcatenationCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class HasValuesTest {
 
     @Test
-    public void contains() {
+    public void matches() {
         MatcherAssert.assertThat(
-            new HasValues<>("a", "b").matchesSafely(
-                new ListOf<>("a", "b", "c", "e"),
-                new StringDescription()
-            ),
-            new IsEqual<>(true)
+            "The matcher check that [1,2,3] contains [2]",
+            new ListOf<>(1, 2, 3),
+            new HasValues<>(2)
         );
     }
 
-    /**
-     * Verify the matcher message in case when test failed.
-     * @todo #32:30m Matcher to test {@link org.hamcrest.Matcher} objects.
-     *  In order to check the matchers we should verify that
-     *  - error message is correct in case false status
-     *  - the status of testing is correct itself.
-     *  The possible name - {@code IsMatcherOf}.
-     */
     @Test
-    public void containsErrorMessage() {
-        final Description msg = new StringDescription();
+    public void contains() {
         MatcherAssert.assertThat(
+            "The matcher check that [a,b,c,e] contains [a,b]",
             new HasValues<>("a", "b").matchesSafely(
-                new ListOf<>("d", "e", "f"),
-                msg
-            ),
-            new IsEqual<>(false)
-        );
-        MatcherAssert.assertThat(
-            msg.toString(),
-            new IsEqual<>("The function applied to <[d, e, f]> is failed")
-        );
-    }
-
-    @Test
-    public void startsWith() {
-        MatcherAssert.assertThat(
-            new HasValues<String>(
-                expected -> expected.startsWith("aa")
-            ).matchesSafely(
-                new ListOf<>("aaA", "bbB", "ccC"),
+                new ListOf<>("a", "b", "c", "e"),
                 new StringDescription()
             ),
             new IsEqual<>(true)
         );
     }
 
+    /**
+     * Once test is failed the hamcrest throw the {@link AssertionError}.
+     * The error has a message with expected/actual results.
+     *
+     * This test is required to check that the result hamcrest message
+     * has proper "actual section". For example the test
+     * <pre>{@code
+     * MatcherAssert.assertThat(
+     *    new ListOf<>(1, 2, 3),
+     *    new HasValues<>(5)
+     * );
+     * }</pre> will fail with the following message
+     * <pre>{@code
+     * java.lang.AssertionError:
+     *   Expected: <[5]>
+     *   but: <[1, 2, 3]>
+     * }</pre>, where
+     *  - the "actual section" is "<[1, 2, 3]>"
+     *  - the "expected section" is "<[5]>".
+     */
     @Test
-    public void nonNull() {
+    public void thatActualSectionOfHamcrestResultMessageIsCorrect() {
```

Comment:
@dgroup I think this one can be called `describeActualValues`

Reply:
@llorllale done.